### PR TITLE
Add GCD and LCM numeric functions

### PR DESCRIPTION
**Pull request by kryonix**
This pr implements functions `gcd` and `lcm`

```
SELECT gcd(42, 57)
3
SELECT lcm(42, 57)
798
```

### CODE REVIEW

**Comment by Alex-Monahan** [comment]
Howdy and thank you!

Would you be open to also adding aliases for these functions that are longer but more descriptive?

Maybe `lowest_common_denom` and `greatest_common_multiple`?

**Comment by kryonix** [comment]
Hey @Alex-Monahan

> Would you be open to also adding aliases for these functions that are longer but more descriptive?

Great idea! I'm all for longer and more descriptive names for these functions.
Added some just now.

File: src/function/scalar/math/numeric.cpp [hunk 1]
```diff
@@ -1161,4 +1162,98 @@ void EvenFun::RegisterFunction(BuiltinFunctions &set) {
...
+void GCDFun::RegisterFunction(BuiltinFunctions &set) {
+	ScalarFunctionSet funcs("gcd");
+
+	funcs.AddFunction(ScalarFunction({LogicalType::TINYINT, LogicalType::TINYINT}, LogicalType::TINYINT,
+	                                 ScalarFunction::BinaryFunction<int8_t, int8_t, int8_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::SMALLINT, LogicalType::SMALLINT}, LogicalType::SMALLINT,
+	                                 ScalarFunction::BinaryFunction<int16_t, int16_t, int16_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::INTEGER, LogicalType::INTEGER}, LogicalType::INTEGER,
+	                                 ScalarFunction::BinaryFunction<int32_t, int32_t, int32_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::BIGINT, LogicalType::BIGINT}, LogicalType::BIGINT,
```
Comment (Mytherin) [inline src/function/scalar/math/numeric.cpp:1211]: Type specialisations for smaller types  are only required for optimization purposes - as the function itself is likely to be quite slow adding another cast is likely not going to be a big cost. Could we remove the `TINYINT, SMALLINT and INTEGER` specializations?

File: src/function/scalar/math/numeric.cpp [hunk 2]
```diff
@@ -1161,4 +1162,98 @@ void EvenFun::RegisterFunction(BuiltinFunctions &set) {
...
+			throw OutOfRangeException("lcm value is out of range");
+		}
+		return TryAbsOperator::Operation<TR, TR>(result);
+	}
+};
+
+void LCMFun::RegisterFunction(BuiltinFunctions &set) {
+	ScalarFunctionSet funcs("lcm");
+
+	funcs.AddFunction(ScalarFunction({LogicalType::TINYINT, LogicalType::TINYINT}, LogicalType::TINYINT,
```
Comment (Mytherin) [inline src/function/scalar/math/numeric.cpp:1243]: Same here - could we remove the specializations for small types?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. Some comments:

File: src/function/scalar/math/numeric.cpp (hunk 1 above)
Comment (kryonix) [reply src/function/scalar/math/numeric.cpp]: Sounds good to me. I removed the specializations for both GCD and LCM.

File: src/function/scalar/math/numeric.cpp (hunk 2 above)
Comment (kryonix) [reply src/function/scalar/math/numeric.cpp]: Done

**Pull request merged**